bemb/model/bemb.py

```diff
@@ -1298,7 +1316,17 @@ def elbo(self, batch: ChoiceDataset, num_seeds: int = 1) -> torch.Tensor:
         # ==============================================================================================================
         # 1. sample latent variables from their variational distributions.
         # ==============================================================================================================
-        sample_dict = self.sample_coefficient_dictionary(num_seeds)
+        if self.deterministic_variational:
+            num_seeds = 1
+            # Use the means of variational distributions as the sole deterministic MC sample.
```

Comment:
this deterministic "sampling" procedure might not be needed. Whenever we pass `forward(deterministic=True)`, the forward function will create a copy of `sample_dict` like this.

Reply:
we need this to send the `sample_dict` to the log_prior